package/gluon-mesh-babel/src/respondd.c

```diff
@@ -989,11 +992,19 @@ static struct json_object * get_wifi(void) {
 
 static struct json_object * respondd_provider_neighbours(void) {
 	struct json_object *ret = json_object_new_object();
+	struct json_object *neighbours_copy = json_object_new_object();
+
+	if (!ret)
+		return NULL;
+
+	if (neighbours_copy) {
+		if (neighbours) {
+			pthread_rwlock_rdlock(&neighbours_lock);
+			json_object_deep_copy(neighbours, neighbours_copy, 0);
```

Comment:
Does this compile? `json_object_deep_copy()` should have the signature `int json_object_deep_copy(struct json_object *src, struct json_object **dst, json_c_shallow_copy_fn *shallow_copy)`.
Also
> The destination object must be initialized to NULL

Reply:
My jenkins says it compiles but I was not yet able to verify the result. In any case, I adjusted the code after reviewing the documentation, considering the return code of json_obect_deep_copy().